### PR TITLE
Fix : charAt of undefined

### PR DESCRIPTION
**Pull request by cristianele**

### CODE REVIEW

**Comment by cvazac** [comment]
thanks for the contribution @cristianele! 

is it possible for you to show us the code where `isNativeElement()` is failing? or, in the case you're seeing, is `typeof nodeName === 'undefined'`?

**Comment by cristianele** [comment]
I can share a screenshot:
screen shot 2017-05-09 at 2 07 58 pm

It's running over NodeJS v6.9.4 with ESlint v3.14.0

**Comment by cvazac** [comment]
I'd really like to get to the bottom of this. If you're willing, can you make this change and give me the output? It will reveal a very minimal amount of source code.

replace this line:
https://github.com/cvazac/eslint-plugin-react-perf/blob/master/lib/utils/node.js#L4

with:
```
  let nodeName
  try {
    nodeName = node.parent.name.name
  } catch (e) {
    var p = node
    while (!p.tokens) {
      p = p.parent
    }

    var source = p.tokens.reduce(function(array, {value, start, end}) {
      if (start <= node.parent.end && end >= node.parent.start)
        array.push(value)
      return array
    }, []).join('')
    console.info('SOURCE**', source)
  }
```

**Comment by cristianele** [comment]
You missed charAt into try:
```
    let nodeName;
    let firstChar;

    try {
        nodeName = node.parent.name.name;
        firstChar = nodeName.charAt(0);
    } catch (e) {
        var p = node
        while (!p.tokens) {
            p = p.parent
        }

        var source = p.tokens.reduce(function(array, {value, start, end}) {
            if (start <= node.parent.end && end >= node.parent.start)
                array.push(value)
            return array
        }, []).join('')
        console.info('SOURCE**', source)
    }
```

Output:
```
SOURCE** (<Form.Inputkey={key}type="checkbox"name={this.props.name}id={`${this.props.name}_${option.value}`}label={option.text}value={option.value}onChange={this.bindings.onCheckBoxChange}labelSize="16px"checkboxSize="28px"/>)
```
The issue comes with components named like Form.Input

File: lib/utils/node.js [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 function isNativeElement (node) {
   const nodeName = node.parent.name.name
```
Comment (cvazac) [inline lib/utils/node.js]: Can you change this to:
```
if (node.parent.name.type !== 'JSXIdentifier') {
  return false;
}
```

**Comment by cvazac** [comment]
thanks for that debugging info! can you add at least one test please? an `invalid` entry here would be good:
https://github.com/cvazac/eslint-plugin-react-perf/blob/master/tests/lib/rules/jsx-no-new-object-as-prop.js#L30

**Comment by cristianele** [comment]
Works perfect!

**Comment by cvazac** [comment]
thanks @cristianele!

**Pull request merged**